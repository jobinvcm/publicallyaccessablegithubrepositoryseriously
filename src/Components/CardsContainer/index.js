import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import { Link } from "react-router-dom";
import propTypes from "prop-types";
import MovieOverview from "../MovieOverview";

const styles = theme => {
  return {
    link: {
      textDecoration: "none"
    },
    gridItem: {
      paddingLeft: "8px"
    },
    gridContainer: {
      position: "relative",
      zIndex: "10"
    }
  };
};

const RenderComponent = (ItemComponent, item) => {
  return <ItemComponent item={item} />;
};

const CardsContainer = props => {
  const { classes, data, ItemComponent } = props;
  if (data.length) {
    return (
      <Grid container spacing={24} className={classes.gridContainer}>
        {data.map(item => {
          return (
            <Grid
              className={classes.gridItem}
              item
              xs={6}
              sm={4}
              lg={2}
              key={item.id}
            >
              <Link className={classes.link} to={`/movie/${item.id}`}>
                {RenderComponent(ItemComponent, item)}
              </Link>
            </Grid>
          );
        })}
      </Grid>
    );
  } else {
    return <div>No Movie Found !!</div>;
  }
};

CardsContainer.propTypes = {
  classes: propTypes.object,
  type: propTypes.string,
  ItemComponent: propTypes.func
};

CardsContainer.defaultProps = {
  type: "movie",
  ItemComponent: MovieOverview
};

export default withStyles(styles)(CardsContainer);
