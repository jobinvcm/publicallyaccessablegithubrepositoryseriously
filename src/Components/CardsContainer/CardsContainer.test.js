import React from 'react';
import { shallow } from 'enzyme';
import CardsContainer from '.';

const data = {
  0: {
    title: "test title",
    release_date: "December 2016",
    imageUrl: "http://testimage",
    vote_average: 4.5
  }
}

describe('<CardsContainer />', () => {
  it('renders CardsContainer component', () => {
    const wrapper = shallow(<CardsContainer data={data}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
