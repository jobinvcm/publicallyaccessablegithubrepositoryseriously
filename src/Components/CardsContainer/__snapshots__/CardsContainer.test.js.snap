// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsContainer /> renders CardsContainer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(CardsContainer)
    data={
      Object {
        "0": Object {
          "imageUrl": "http://testimage",
          "release_date": "December 2016",
          "title": "test title",
          "vote_average": 4.5,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "ItemComponent": [Function],
      "classes": Object {
        "gridContainer": "CardsContainer-gridContainer-3",
        "gridItem": "CardsContainer-gridItem-2",
        "link": "CardsContainer-link-1",
      },
      "data": Object {
        "0": Object {
          "imageUrl": "http://testimage",
          "release_date": "December 2016",
          "title": "test title",
          "vote_average": 4.5,
        },
      },
      "type": "movie",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "ItemComponent": [Function],
        "classes": Object {
          "gridContainer": "CardsContainer-gridContainer-3",
          "gridItem": "CardsContainer-gridItem-2",
          "link": "CardsContainer-link-1",
        },
        "data": Object {
          "0": Object {
            "imageUrl": "http://testimage",
            "release_date": "December 2016",
            "title": "test title",
            "vote_average": 4.5,
          },
        },
        "type": "movie",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
